lib/hooks.js

```diff
@@ -0,0 +1,91 @@
+'use strict'
+
+const lifecycleHooks = ['preParsing', 'preValidation', 'preExecution', 'preGatewayExecution', 'onResolution']
+const supportedHooks = lifecycleHooks
+const { MER_ERR_HOOK_INVALID_TYPE, MER_ERR_HOOK_INVALID_HANDLER, MER_ERR_HOOK_UNSUPPORTED_HOOK } = require('./errors')
+
+function Hooks () {
+  this.preParsing = []
+  this.preValidation = []
+  this.preExecution = []
+  this.preGatewayExecution = []
+  this.onResolution = []
+}
+
+Hooks.prototype.validate = function (hook, fn) {
+  if (typeof hook !== 'string') throw new MER_ERR_HOOK_INVALID_TYPE()
+  if (typeof fn !== 'function') throw new MER_ERR_HOOK_INVALID_HANDLER()
+  if (supportedHooks.indexOf(hook) === -1) {
```

Comment:
```suggestion
  if (lifecycleHooks.indexOf(hook) === -1) {
```

I think you can remove the `supportedHooks` variable